spec/grover/middleware_spec.rb

```diff
@@ -860,6 +860,46 @@
       end
     end
 
+    describe '#ignore_request?' do
+      context 'when app configuration has a Proc for ignore_request' do
+        context 'with a hostname' do
+          before { allow(Grover.configuration).to receive(:ignore_request).and_return(->(req) { req.host == 'www.example.org' }) }
+
+          it 'request is ignored when the request path passed to the proc defined in ignore_request returns true' do
+            response = get 'http://www.example.org/foobazbar'
+            expect(response.content_type).to eq 'text/html'
+
+            response = get 'http://www.example.org/foobazbar.pdf'
+            expect(response.content_type).to eq 'text/html'
+
+            response = get 'http://www.example.org/foobarbar'
+            expect(response.content_type).to eq 'text/html'
+
+            response = get 'http://www.therealexample.org/foobarbar.pdf'
+            expect(response.content_type).to eq 'application/pdf'
+          end
+        end
+
+        context 'with a custom header' do
+          before { allow(Grover.configuration).to receive(:ignore_request).and_return(->(req) { req.has_header?('X-BLOCK') }) }
+
+          it 'request is ignored when the request path passed to the proc defined in ignore_request returns true' do
+            response = get 'http://www.example.org/foobazbar', {}, { 'X-BLOCK' => '1' }
+            expect(response.content_type).to eq 'text/html'
+
+            response = get 'http://www.example.org/foobazbar.pdf', {}, { 'X-BLOCK' => '1' }
+            expect(response.content_type).to eq 'text/html'
+
+            response = get 'http://www.example.org/foobarbar', {}, { 'X-BLOCK' => '1' }
+            expect(response.content_type).to eq 'text/html'
+
+            response = get 'http://www.therealexample.org/foobarbar.pdf'
```

Comment:
ideally this should match the request on L890 (aside from the "block" header) so as to minimise the things changing between success and failure.

Reply:
OK, I think I got you.

1. blocking with host (html request) returns html
2.blocking with host (pdf request) returns html
3. not blocking with host (html request) returns html
4. not blocking with host (pdf request) returns pdf